test/Interop/Cxx/metadata/mirror.swift

```diff
@@ -0,0 +1,95 @@
+// RUN: %target-run-simple-swift(-cxx-interoperability-mode=default -I %S/Inputs)
```

Comment:
I would probably move this test into `Interop/Cxx/class` or `Interop/Cxx/foreign-reference` to avoid creating a new subdirectory. I guess we probably wouldn't have many more tests for C++ type metadata specifically.